src/view/screens/Search/Search.tsx

```diff
@@ -671,19 +642,27 @@ export function SearchScreen(
   const onSubmit = React.useCallback(() => {
     scrollToTopWeb()
     setShowAutocompleteResults(false)
-    updateSearchHistory(query)
-  }, [query, setShowAutocompleteResults, updateSearchHistory])
+    updateSearchHistory(searchText)
+    setQueryTerm(searchText)
+
+    if (isWeb) {
+      navigation.push('Search', {q: searchText})
+    } else {
+      navigation.setParams({q: searchText})
+    }
+  }, [navigation, searchText, updateSearchHistory])
```

Comment:
As mentioned above, we don't want to push on native because there isn't a back button. Whenever we rework the shell and potentially get rid of the drawer icon, we can revisit.